pkg/boot/linux.go

```diff
@@ -62,7 +63,36 @@ func (li *LinuxImage) String() string {
 	return fmt.Sprintf("LinuxImage(\n  Name: %s\n  Kernel: %s\n  Initrd: %s\n  Cmdline: %s\n)\n", li.Name, stringer(li.Kernel), stringer(li.Initrd), li.Cmdline)
 }
 
-func copyToFile(r io.Reader) (*os.File, error) {
+func copyToFile(r io.Reader, name string) (*os.File, error) {
+	// If source is a regular file in tmpfs, simply re-use that than copy.
```

Comment:
I think I'm missing something. Where's the check that it's a regular file?

Reply:
My understanding is that if `os.Stat` can tell the path exist from file system, it is a file -- regular in a sense it exists, other than need to read / stream from other types, which looks like could be "os.File could be a socket, a pipe or some other strange thing" (I don't know where we would run into such case, realistically)

Reply:
I think the terminology is slightly off. You can stat fifos and sockets like:

```
~$ mkfifo f
~$ stat f
  File: f
  Size: 0         	Blocks: 0          IO Block: 4096   fifo
...
```

Reply:
You are right -- the code currently is missing an additional check, updated the code to be:

```
if fi, err := os.Stat(name); err == nil && fi.Mode().IsRegular() {
   ...
}
```